org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XSISchemaModel.java

```diff
@@ -102,4 +124,45 @@ private static void createCompletionItem(String text, boolean isSnippetsSupporte
 	private static boolean attributeAlreadyExists(DOMElement root, String actualPrefix, String suffix) {
 		return root.getAttributeNode(actualPrefix + ":" + suffix) != null; 
 	}
+
+	public static Hover computeHoverResponse(DOMAttr attribute, IHoverRequest request) {
+
+		String name = attribute.getName();
+		if(!name.startsWith(request.getXMLDocument().getSchemaInstancePrefix() + ":")) {
+			return null;
+		}
+
+		DOMDocument document = request.getXMLDocument();
+		DOMElement root = document.getDocumentElement();
+		String doc = null;
+		if(root != null) {
+			if(root.equals(document.findNodeAt(attribute.getStart()))) {
+				if(name.endsWith(":schemaLocation")) {
+					doc = SCHEMA_LOCATION_DOC;
+				}
+				else if(name.endsWith(":noNamespaceSchemaLocation")) {
+					doc = NO_NAMESPACE_SCHEMA_LOCATION_DOC;
+				}
+			}
+		} else {
+			return null;
+		}
+		if(doc == null) {
+			if(name.endsWith(":nil")) {
+				doc = NIL_DOC;
+			}
+			else if(name.endsWith(":type")) {
+				doc = TYPE_DOC;
+			}
+			else {
+				return null;
+			}
+		}
+		
+
+		MarkupContent content = new MarkupContent();
+		content.setKind(MarkupKind.MARKDOWN);
```

Comment:
what bothers me here is the inconsistency between this hover being markdown, others being plaintext. It's probably tolerable but we should probably try to return markdown for other documentation. That's the subject of another ticket though.

Reply:
Opened #245